### DISCUSSION TITLE
How do I add self-hosted Ruffle onto a Google Site?

### DISCUSSION THREAD

**Discussion by Spaghettiboi67** [ruffle-rs/ruffle#18901] [Q&A]
For the last couple of days, I've been trying to figure out how I can host Ruffle on a Google Site so they can upload their own .swf files and play from there. How do I do that?

**Comment by n0samu** [comment]
The new Google Sites is not a good option for creating game sites. You could add Ruffle via the CDN (see the script tag example [on our website](https://ruffle.rs/downloads#website-package)). But you will still need some other place to upload and host your SWF files, because the new Google Sites doesn't allow uploading arbitrary files anymore.

**Reply by Spaghettiboi67** [reply]
Ok so I can't use Google Sites. If not, I need another website creator using javascript, which do you reccomend?

**Reply by n0samu** [reply]
I recommend creating your site with GitHub Pages: https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site

**Reply by Spaghettiboi67** [reply]
What if I can't use .io pages?
